src/wire_protocol/operations.rs

```diff
@@ -46,7 +30,7 @@ pub enum Message {
         /// The total number of documents being returned.
         number_returned: i32,
         /// The documents being returned.
-        documents: Vec<bson::Document>,
+        documents: Vec<u8>,
```

Comment:
This could alternatively be a `Vec<Vec<u8>>`, where each Vec<u8> represents a single document.  The first four bytes of each document are the length of the document, so it's fairly easy to split.